verifier/trustpolicy/blob.go

```diff
@@ -0,0 +1,133 @@
+}
+
+// BlobTrustPolicy represents a policy statement in the blob policy document
+type BlobTrustPolicy struct {
+	// Name of the policy statement
+	Name string `json:"name"`
+
+	// SignatureVerification setting for this policy statement
+	SignatureVerification SignatureVerification `json:"signatureVerification"`
+
+	// TrustStores this policy statement uses
+	TrustStores []string `json:"trustStores"`
+
+	// TrustedIdentities this policy statement pins
+	TrustedIdentities []string `json:"trustedIdentities"`
+
+	// GlobalPolicy defines if policy statement is global or not
+	GlobalPolicy bool `json:"globalPolicy,omitempty"`
+}
+
+// LoadBlobDocument loads a trust policy document from a local file system
+func LoadBlobDocument() (*BlobDocument, error) {
+	var doc BlobDocument
+	err := getDocument(dir.PathBlobTrustPolicy, &doc)
+	return &doc, err
+}
+
+// Validate validates a policy document according to its version's rule set.
+// if any rule is violated, returns an error
+func (policyDoc *BlobDocument) Validate() error {
+	// sanity check
+	if policyDoc == nil {
+		return errors.New("blob trust policy document cannot be nil")
+	}
+
+	// Validate Version
+	if policyDoc.Version == "" {
+		return errors.New("blob trust policy has empty version, version must be specified")
+	}
+	if !slices.Contains(supportedPolicyVersions, policyDoc.Version) {
+		return fmt.Errorf("blob trust policy document uses unsupported version %q", policyDoc.Version)
+	}
+
+	// Validate the policy according to 1.0 rules
+	if len(policyDoc.BlobTrustPolicies) == 0 {
+		return errors.New("blob trust policy document can not have zero trust policy statements")
+	}
+
+	policyNames := set.New[string]()
+	foundGlobalPolicy := false
+	for _, statement := range policyDoc.BlobTrustPolicies {
+		// Verify unique policy statement names across the policy document
+		if policyNames.Contains(statement.Name) {
+			return fmt.Errorf("multiple blob trust policy statements use the same name %q, statement names must be unique", statement.Name)
+		}
+		policyNames.Add(statement.Name)
+
+		if err := validatePolicyCore(statement.Name, statement.SignatureVerification, statement.TrustStores, statement.TrustedIdentities); err != nil {
+			return fmt.Errorf("blob trust policy: %w", err)
+		}
+
+		if statement.GlobalPolicy {
+			if foundGlobalPolicy {
+				return errors.New("multiple blob trust policy statements have globalPolicy set to true. Only one trust policy statement should be marked as global policy")
+			}
+			foundGlobalPolicy = true
+		}
+	}
+
+	return nil
+}
+
+// GetApplicableTrustPolicy returns a pointer to the deep copied TrustPolicy
+// see https://github.com/notaryproject/notaryproject/blob/v1.1.0/specs/trust-store-trust-policy.md#blob-trust-policy
+func (policyDoc *BlobDocument) GetApplicableTrustPolicy(policyName string) (*BlobTrustPolicy, error) {
+	for _, policyStatement := range policyDoc.BlobTrustPolicies {
+		if policyName == "" {
```

Comment:
How can we make sure it is the user's intent to apply the global policy and not unintentionally picking it because they made an error in passing the policyName? At notation CLI layer, picking the global policy if PolicyName is empty makes sense. But at notation-go layer, should we make sure the customer is intentional in picking global policy? I can see a user shooting themselves in the foot by picking the global policy because their code has a bug where it is not properly initializing the policyName variable.

Reply:
Done, made changes for blob policy.